datalad/support/annexrepo.py

```diff
@@ -2160,6 +2160,25 @@ def _run_annex_command_json(self, command,
                         "Received no json output for --json command, only:\n{}"
                         .format("  ".join(others)))
                 raise exc
+
+        # A special remote might send a message via "info". This is supposed to be printed by annex but in case of
+        # `--json` is returned by annex as "{'info': '<message>'}".
+        # See https://git-annex.branchable.com/design/external_special_remote_protocol/#index5h2
+        #
+        # So, Ben thinks we should just spit it out here, since everything calling _run_annex_command_json
+        # is concerned with the actual results being returned. More over, this kind of response is special to particular
+        # special remotes rather than particular annex commands. So, likely there's nothing callers could do about it
+        # other than spitting it out.
+
+        to_remove = []
+        for obj in json_objects:
+            if len(obj.keys()) == 1 and obj['info']:
+                lgr.info(obj['info'])
+                to_remove.append(obj)
+
+        for obj in to_remove:
+            json_objects.remove(obj)
```

Comment:
Code will be shorter if you just compose that list to return in the loop... Not sure even if remove would work correctly here eh in case of multiple identical info messages